### DISCUSSION TITLE
https://nextjs.org/learn/basics/assets-metadata-css/layout-component error

### DISCUSSION THREAD

**Discussion by yangyangnote** [vercel/next.js#53657] [Help] [answered]
### Summary

When I was studying
https://nextjs.org/learn/basics/assets-metadata-css/layout-component
1.First, let's create a layout component that will be shared across all pages
2.Also wrote this code
	```
	  export default function Layout({ children }) {
	    return <div>{children}</div>;
	  }
	```
3.. Then, open pages/posts/first-post.js and add an import for the layout component and make it the outermost component

	```
	  import Head from 'next/head';
	  import Link from 'next/link';
	  import Layout from '../../components/layout';
	  - export default function FirstPost() {
	  return (
	    <Layout>
	      <Head>
	        <title>First Post</title>
	      </Head>
	      <h1>First Post</h1>
	      <h2>
	        <Link href="/">← Back to home</Link>
	      </h2>
	    </Layout>
	  );
	  }
	```
4.I also created components/layout.module.css

	```
	  .container {
	    max-width: 36rem;
	    padding: 0 1rem;
	    margin: 3rem auto 6rem;
	  }
	```
5.and used in the project
	```
	  import styles from './layout.module.css';
	  
	  export default function Layout({ children }) {
	    return <div className={styles.container}>{children}</div>;
	  }
	```
When I visit [http://localhost:3000/posts/first-post](http://localhost:3000/posts/first-post)
	#### Failed to compile
	```
	  ./components/layout.js:1:0
	  **Module not found**: Can't resolve './layout.module.css'
	  **>** 1 | import styles from './layout.module.css';
	  2 | 
	  3 | export default function Layout({ children }) {
	  4 |   return <div className={styles.container}>{children}</div>;
	  - https://nextjs.org/docs/messages/module-not-found
	  - Import trace for requested module:
	  ./pages/posts/first-post.js
	```

why 😭

**Comment by icyJoseph** [comment]
Well, the root issue is that from wherever step 5 is, you cannot find `layout.module.css`. Could you share a screenshot or printout of the file structure you are using? 

In other words, where is step 4, relative to step 5? 

The way you have written it, step 5, has to be a sibling to step 4, in other words, they must be under the same directory parent.

Also, depending on your operating system, if you have named step 4 as: `Layout.module.css` it won't work as expected, it has to be all lowercased.

**Reply by yangyangnote** [reply]
1.Thank you for your help, my file structure looks like this

![CleanShot 2023-08-08 at 15 57 40](https://github.com/vercel/next.js/assets/19338570/c6dd44d8-5f9b-4a58-98f9-07a872a6dc90)


2.My layout.module.css looks like this
![CleanShot 2023-08-08 at 15 53 05](https://github.com/vercel/next.js/assets/19338570/3d4a385c-b8bf-4bc9-809a-fc8eb17275ef)


3.The code in my layout.js looks like this
![CleanShot 2023-08-08 at 15 53 30](https://github.com/vercel/next.js/assets/19338570/d1dd6d37-f810-4fb9-8abe-3ee8d1db1fbb)

There doesn't seem to be the problem you mentioned。
Or maybe there's something else going on that I'm not seeing ？😄

**Reply by icyJoseph** [reply]
There doesn't seem to be anything wrong.

What next.js version have you got?

Would it be possible for you to create a repository on GitHub?

**Reply by yangyangnote** [reply]
Thank you very much for your help😄
my next.js version is next@13.4.12

I create a new repository ,address
https://github.com/yangyangnote/nextjsblog

**Reply by icyJoseph** [reply, marked as answer]
Hi,

This will sound a bit weird, but your file name `layout.module.css ` is incorrect, because there's a whitespace at the end of the name.

- You have: `layout.module.css `
- Should be: `layout.module.css`

**Reply by yangyangnote** [reply]
😭,Thank you very much for pointing out the error, which is really hard to find!

**Reply by icyJoseph** [reply]
@yangyangnote no worries! Please don't forget to mark my comment as an answer :) have a good day

**Reply by yangyangnote** [reply]
OK 👌
